etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestrator.java

```diff
@@ -33,15 +35,41 @@ private PartnerMetadataOrchestrator() {}
 
     public void updateMetadataForReceivedOrder(String receivedSubmissionId, Order<?> order)
             throws PartnerMetadataException {
-        // will call the RS history API given the submissionId, currently blocked by:
-        // https://github.com/CDCgov/prime-reportstream/issues/12624
-        // from the response, extract the "sender" and "timestamp" (timeReceived)
-        // we will calculate the hash.
-        // then we call the metadata storage to save this stuff.
-
-        String sender = "unknown";
-        Instant timeReceived = Instant.now();
-        String hash = String.valueOf(order.hashCode());
+        // currently blocked by: https://github.com/CDCgov/prime-reportstream/issues/12624
+        // once we get the right receivedSubmissionId from RS, this method should work
+        logger.logInfo(
+                "Looking up sender name and timeReceived from RS history API for receivedSubmissionId: {}",
+                receivedSubmissionId);
+
+        String sender;
+        Instant timeReceived;
+        String hash;
+        try {
+            String bearerToken = rsclient.getRsToken();
+            String responseBody =
+                    rsclient.requestHistoryEndpoint(receivedSubmissionId, bearerToken);
+            Map<String, Object> responseObject =
+                    formatter.convertJsonToObject(responseBody, new TypeReference<>() {});
+
+            var senderObj = responseObject.get("sender");
```

Comment:
Same here as on the other, I'd just make these Strings

Reply:
Done